_scripts/app2-10.R

```diff
@@ -0,0 +1,82 @@
+{
+  
+  # Application 2-10 ----
+  
+  rm(list=ls());
+  options(show.error.locations = TRUE);
+  
+  ## Part 1 ----
+  
+  counter <- function(vector, compareVal, conditionalOp = ">"){
+    vecLength <- length(vector); # length of vector
+    conditionMet <- FALSE;
+    
+    for(i in 1:vecLength){
+      if(conditionalOp != ">" && conditionalOp != "<" &&
+         conditionalOp != ">=" && conditionalOp != "<=" &&
+         conditionalOp != "==" && conditionalOp != "!="){
+        cat("Error! Invalid Conditional Operator!")
+      }
+      else if(conditionalOp == ">" && vector[i] > compareVal){
```

Comment:
Technically correct but not efficient.  You have six **_else if_** conditions that all lead to the same code (**_conditionMet = TRUE_**).  Change this so that there is only one **_else if_** condition that does the exact same thing as the 6 in your code.  It will be a much larger condition.